torch/csrc/jit/codegen/cuda/ir_nodes.cpp

```diff
@@ -206,6 +206,33 @@ bool ReductionOp::sameAs(const ReductionOp* other) const {
       this->init()->sameAs(other->init()));
 }
 
+std::vector<IterDomain*> ReductionOp::getReductionDomains() const {
+  const Val* out_val = out();
+  // out is a TensorIndex after lowering
+  if (out_val->getValType() == ValType::TensorIndex) {
+    out_val = static_cast<const TensorIndex*>(out_val)->view();
+  }
+  auto vec_domain = out_val->as<TensorView>()->domain()->domain();
+  vec_domain.erase(
+      std::remove_if(
+          vec_domain.begin(),
+          vec_domain.end(),
+          [](IterDomain* id) { return !id->isReduction(); }),
+      vec_domain.end());
+  return vec_domain;
+}
+
+std::unordered_map<ParallelType, IterDomain*> ReductionOp::
```

Comment:
That's certainly possible and would be more efficient. However, we also need to create a mapping of the six parallel type values to [0, 6). Since the six types are indeed the first six values defined in `enum ParallelType`, we could just cast them to int, but this seems to create an additional implicit dependency. I think I'm in favor of just using `unordered_map` for simplicity.